classes/Tab.php

```diff
@@ -137,11 +145,14 @@ public static function initAccess($idTab, Context $context = null)
         $slug = 'ROLE_MOD_TAB_' . strtoupper(self::getClassNameById($idTab));
 
         foreach (array('CREATE', 'READ', 'UPDATE', 'DELETE') as $action) {
-            Db::getInstance()->execute('INSERT INTO `' . _DB_PREFIX_ . 'authorization_role` (`slug`) VALUES ("' . $slug . '_' . $action . '")');
+            //Check if authorization role does not exist (this can happen if you want to create several tabs with the same class_name or route_name)
```

Comment:
Maybe more readable?
```php
/**
 * Check if authorization role does not exist.
 * This can happen if you want to create several tabs with
 * the same class_name or route_name.
 */
$slug = pSQL($slug . '_' . $action);
$authorizationRoleExist = Db::getInstance()->numRows(
    'SELECT 1 FROM `' . _DB_PREFIX_ . 'authorization_role` ' .
    'WHERE `slug` = "' . $slug . '"'
);

if ($authorizationRoleExist === 0) {
    Db::getInstance()->execute(
        'INSERT INTO `' . _DB_PREFIX_ . 'authorization_role` (`slug`) VALUES ("' . $slug . '")'
    );
}

```